### PR TITLE
add glFilter and mutex

### PR DESCRIPTION
**Pull request by Goaman**

### CODE REVIEW

File: addons/html_builder/static/src/builder/builder_helpers.js [hunk 1]
```diff
@@ -138,7 +138,63 @@ const actionsRegistry = registry.category("website-builder-actions");
 export function useClickableWeWidget() {
     useWeComponent();
     const comp = useComponent();
-    const call = comp.env.editor.shared.history.makePreviewableOperation(callActions);
+    const _call = comp.env.editor.shared.history.makePreviewableOperation(callActions);
```
Comment (FrancoisGe) [inline addons/html_builder/static/src/builder/builder_helpers.js]: _call => applyOperation ? applyAction ?

File: addons/html_builder/static/src/builder/builder_helpers.js [hunk 2]
```diff
@@ -138,68 +139,83 @@ const actionsRegistry = registry.category("website-builder-actions");
...
+        comp.env.editor.shared.operation.next(
+            () => {
+                fn(shorthandApplyDefs.concat(customApplyDefs));
+            },
+            {
+                load: async () => {
+                    const customAction = getCustomAction();
+                    const customActionLoad =
+                        customAction?.actionId && actionsRegistry.get(customAction.actionId).load;
+                    if (!customActionLoad) {
```
Comment (FrancoisGe) [inline addons/html_builder/static/src/builder/builder_helpers.js]: we force 1 microTick all the time ?

File: addons/html_builder/static/src/builder/builder_helpers.js [hunk 3]
```diff
@@ -138,68 +139,83 @@ const actionsRegistry = registry.category("website-builder-actions");
...
+                });
+            }
+        }
+        return shorthandOperations;
+    }
+
+    const call = (fn, operationParams) => {
+        const shorthandApplyDefs = getOperations(getShorthandActions());
+        const customAction = getCustomAction();
+        const customApplyDefs = (customAction && getOperations([customAction])) || [];
```
Comment (FrancoisGe) [inline addons/html_builder/static/src/builder/builder_helpers.js]: applyDefs ?

File: addons/html_builder/static/src/builder/builder_helpers.js [hunk 4]
```diff
@@ -138,68 +139,83 @@ const actionsRegistry = registry.category("website-builder-actions");
...
+                    const customActionLoad =
+                        customAction?.actionId && actionsRegistry.get(customAction.actionId).load;
+                    if (!customActionLoad) {
                         return;
                     }
-                    fn(loadResult);
-                });
-            })
+                    const results = await Promise.all(
+                        customApplyDefs.map((applyDef) =>
```
Comment (FrancoisGe) [inline addons/html_builder/static/src/builder/builder_helpers.js]: ???

File: addons/html_builder/static/src/builder/builder_helpers.js [hunk 5]
```diff
@@ -138,68 +139,83 @@ const actionsRegistry = registry.category("website-builder-actions");
...
+                                param: applyDef.actionParam,
+                                value: applyDef.actionValue,
+                                // todo: should not be necessary if the actions are registred
+                                // through a plugin resource
+                                editor: comp.env.editor,
+                                loadResult: applyDef.loadResult,
+                            })
+                        )
+                    );
+                    return zip(customApplyDefs, results).map(([applyDef, result]) => {
```
Comment (FrancoisGe) [inline addons/html_builder/static/src/builder/builder_helpers.js]: ???

File: addons/html_builder/static/src/builder/builder_helpers.js (hunk 2 above)
Comment (Goaman) [reply addons/html_builder/static/src/builder/builder_helpers.js]: yep

**Pull request merged**